### ISSUE TITLE
t/examples.t fails with -Duselongdouble

### ISSUE THREAD

**Issue by eserte** [FGasper/p5-CBOR-PP#1]
If perl was compiled with increased floating point accuracy (i.e.with `-Duselongdouble`) then the test suite fails:
```
#   Failed test 'Decode fb3ff199999999999a'
#   at t/examples.t line 134.
#          got: '1.10000000000000009'
#     expected: '1.1'
# 1.10000000000000009

#   Failed test 'Decode fbc010666666666666'
#   at t/examples.t line 134.
#          got: '-4.09999999999999964'
#     expected: '-4.1'
# -4.09999999999999964
# Looks like you failed 2 tests of 151.
t/examples.t ......... 
Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/151 subtests 
```

**Comment by FGasper** [comment]
You know, I wonder if this is a bug in Perl. The underlying C logic is able to typecast from long double to plain double; maybe that’s not being done here?

-F

**Comment by eserte** [comment]
It's only partially fixed. Amongst my smoker systems I have three perl configurations with -Duselongdouble, two of them are fixed now (a freebsd 10 system with perl 5.20.1 and a debian/jessie system with perl 5.22.3). However, there's still an error with on a freebsd 12 system with perl 5.28.1:
```
#   Failed test 'we decode what CBOR::XS encoded'
#   at t/against_cbor_xs.t line 78.
#          got: '1.10000000000000009'
#     expected: '1.1'
# 1.10000000000000009
# Looks like you failed 1 test of 58.
t/against_cbor_xs.t .. 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/58 subtests 
```

**Comment by FGasper** [comment]
Doh, I missed one.

The 0.03 release should address this; at least, I repro’d a problem with a local long-double Perl and confirmed the fix with 0.03. I’ll leave this issue open pending confirmation from you.

**Comment by eserte** [comment]
My three *-ld configurations all reported passes with 0.03:
```
$ ls -al *.CBOR-PP-0.03*ld*
-rw-r--r-- 1 eserte eserte 7597 Mai 12 20:49 pass.CBOR-PP-0.03.amd64-freebsd-ld.10.1-release.1557686938.98739.rpt
-rw-r--r-- 1 eserte eserte 7013 Mai 13 21:58 pass.CBOR-PP-0.03.amd64-freebsd-ld.12.0-stable.1557777518.91646.rpt
-rw-r--r-- 1 eserte eserte 7603 Mai 12 21:27 pass.CBOR-PP-0.03.x86_64-linux-ld.3.16.0-4-amd64.1557689230.30315.rpt
```
